src/NewsFeed/NewsFeed.tsx

```diff
@@ -38,79 +52,92 @@ class NewsFeed extends React.Component<
     return result;
   }
 
-  componentDidMount() {
+  useEffect(() => {
     /* besides stories, occasionally, there are a poll or two */
     fetch("https://hacker-news.firebaseio.com/v0/topstories.json")
       .then((response) => response.json())
       .then(
         (data) => {
-          this.setState({
-            randomStories: this.getRandom(data, 10),
-            hasLoaded: 1,
+          const randomStoryIds: string[] = getRandom(data, 10); // Assuming getRandom returns an array of story IDs as strings
+
+          // Construct URLs from story IDs
+          storyUrls = randomStoryIds.map((storyId: string) => {
+            return `https://hacker-news.firebaseio.com/v0/item/${storyId}.json`;
           });
 
-          this.randomStoriesData = this.state.randomStories.map(
-            (storyId: string) => {
-              return `https://hacker-news.firebaseio.com/v0/item/${storyId}.json`;
-            }
-          );
+          setLoading(1);
+          setRandomStories(randomStoryIds);
+          // setRandomStoriesData(storyUrls);
+          randomStoriesData = storyUrls;
         },
         (error) => {
           console.log("Error! Could not get top stories. ", error);
         }
       )
-      .then((randomStoriesData) => {
-        Object.entries(this.randomStoriesData).forEach((value: string[]) => {
-          fetch(value[1])
+      .then(() => {
+        const promises = randomStoriesData.map((story: string) => {
+          return fetch(story)
             .then((response) => response.json())
             .then(
               (data) => {
-                this.randomStoriesContent.push(data);
+                randomStoriesContent.push(data);
+                return randomStoriesContent;
```

Comment:
You don't need to return anything here as this value is not assigned anywhere. Same as below.